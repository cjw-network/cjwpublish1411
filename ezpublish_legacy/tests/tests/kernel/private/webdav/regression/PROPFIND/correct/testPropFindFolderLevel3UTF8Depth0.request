<?php

// Test if PROPFIND returns correct information for the directory $GLOBALS['ezc_webdav_testfolder']
// containing a file with UTF-8 characters in the name
// with HTTP_DEPTH = 0

// First upload the file in the folder $GLOBALS['ezc_webdav_testfolder']
$article = new ezpObject( 'file', (int)$GLOBALS['ezc_webdav_testfolderid'] );
$article->name = "româneşte";
$article->file = dirname( __FILE__ ) . DIRECTORY_SEPARATOR . $GLOBALS['ezc_webdav_testfolder'] . '.txt';
$article->publish();

$_SERVER['REQUEST_URI'] = '/' . $GLOBALS['ezc_siteaccess'] . '/Content/' . $GLOBALS['ezc_webdav_testfolder'] . '/';
$_SERVER['REQUEST_METHOD'] = 'PROPFIND';

$_SERVER['HTTP_DEPTH'] = '0';
$_SERVER['CONTENT_TYPE'] = 'application/xml';
$_SERVER['HTTP_CONTENT_LENGTH'] = '288';

// The PROPFIND request needs an XML body
$GLOBALS['ezc_post_body'] = <<<END
<?xml version="1.0" encoding="utf-8"?>
<propfind xmlns="DAV:"><prop>
<creationdate xmlns="DAV:"/>
<getcontentlength xmlns="DAV:"/>
<getcontenttype xmlns="DAV:"/>
<getlastmodified xmlns="DAV:"/>
<displayname xmlns="DAV:"/>
<executable xmlns="http://apache.org/dav/props/"/>
<resourcetype xmlns="DAV:"/>
<supportedlock xmlns="DAV:"/>
<checked-in xmlns="DAV:"/>
<checked-out xmlns="DAV:"/>
</prop></propfind>
END;

?>
