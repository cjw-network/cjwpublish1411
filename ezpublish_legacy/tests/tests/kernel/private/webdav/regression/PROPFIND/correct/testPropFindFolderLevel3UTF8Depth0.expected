Content-Type: text/xml; charset="utf-8"
Server: eZComponents/dev/ezcWebdavTransportWrapper
HTTP/1.1 207 Multi-Status

<?xml version="1.0" encoding="UTF-8"?>
<D:multistatus xmlns:D="DAV:">
  <D:response xmlns:default="http://apache.org/dav/props/">
    <D:href>http://@ezc_webdav_host@/@ezc_siteaccess@/Content/@ezc_webdav_testfolder@/</D:href>
    <D:propstat>
      <D:prop>
        <D:getcontentlength>4096</D:getcontentlength>
        <D:getlastmodified>Thu, 05 Mar 2009 13:27:53 +0000</D:getlastmodified>
        <D:creationdate>2009-03-05T13:27:53+0000</D:creationdate>
        <D:displayname>@ezc_webdav_testfolder@</D:displayname>
        <D:getcontenttype>httpd/unix-directory</D:getcontenttype>
        <D:resourcetype>
          <D:collection/>
        </D:resourcetype>
      </D:prop>
      <D:status>HTTP/1.1 200 OK</D:status>
    </D:propstat>
    <D:propstat xmlns:default="http://apache.org/dav/props/">
      <D:prop>
        <D:checked-in/>
        <D:checked-out/>
        <default:executable xmlns="http://apache.org/dav/props/"/>
      </D:prop>
      <D:status>HTTP/1.1 404 Not Found</D:status>
    </D:propstat>
  </D:response>
</D:multistatus>
